src/boost_histogram/tag.py

```diff
@@ -185,21 +185,23 @@ def group_mapping(self, axis: PlottableAxis) -> Sequence[int]:
                 newedges = self.axis.edges
 
             if newedges is not None and hasattr(axis, "edges"):
-                assert newedges[0] == axis.edges[0], "Edges must start at first bin"
-                assert newedges[-1] == axis.edges[-1], "Edges must end at last bin"
-                assert all(
+                if newedges[0] != axis.edges[0]:
+                    msg = "Edges must start at first bin"
+                    raise ValueError(msg)
+                if newedges[-1] != axis.edges[-1]:
+                    msg = "Edges must end at last bin"
+                    raise ValueError(msg)
+                matched_ixes = [np.abs(axis.edges - edge).argmin() for edge in newedges]
+                all_close = all(
                     np.isclose(
-                        axis.edges[np.abs(axis.edges - edge).argmin()],
+                        axis.edges[ix],
                         edge,
                     )
-                    for edge in newedges
-                ), "Edges must be in the axis"
-                matched_ixes = np.where(
-                    np.isin(
-                        axis.edges,
-                        newedges,
-                    )
-                )[0]
+                    for ix, edge in zip(matched_ixes, newedges)
+                )
+                if not all_close:
+                    msg = "Edges must be in the axis"
+                    raise ValueError(msg)
```

Comment:
Could we emit the offending edge? It would make for easier debug by users. Something like
```
f"Edge(s) {missing_edges} not found in the axis. Nearest available edges are ..."
```

Reply:
Sure, just added that.

Reply:
Sure, just added that.